source/patterns/@aws-solutions-constructs/aws-dynamodb-stream-lambda/lib/index.ts

```diff
@@ -88,20 +88,24 @@ export class DynamoDBStreamToLambda extends Construct {
       lambdaFunctionProps: props.lambdaFunctionProps
     });
 
-    this.dynamoTable = defaults.buildDynamoDBTableWithStream(this, {
+    const table: dynamodb.ITable = defaults.buildDynamoDBTableWithStream(this, {
       dynamoTableProps: props.dynamoTableProps,
       existingTableObj: props.existingTableObj
     });
 
+    if (table instanceof dynamodb.Table) {
```

Comment:
The definition of `buildDynamoDBTableWithStream` is:

```
export function buildDynamoDBTableWithStream(scope: cdk.Construct, props: BuildDynamoDBTableWithStreamProps): dynamodb.ITable {
```

Typescript sees this and will never trigger this condition (pretty sure instanceof isn't fully functional at runtime as the code actually runs as JavaScript with no type information). Either way, the function will never return a Table.

Reply:
First thought suggestion, define the function as:

```
export function buildDynamoDBTableWithStream(scope: cdk.Construct, props: BuildDynamoDBTableWithStreamProps): [ dynamodb.ITable, dynamodbTable ] {
```
Then return everything you can from the from the function (either ITable, undefined or ITable, Table). Assign both to properties of the construct (and stressing again to explain the difference in the docs). I'm not a fan of this operator to return multiple values, but it seems to work here. There may be a better way.

Reply:
Good catch!